### PR TITLE
Add remove teacher API

### PR DESCRIPTION
**Pull request by karenc**

### CODE REVIEW

File: app/controllers/api/v1/courses_controller.rb [hunk 1]
```diff
@@ -64,6 +64,15 @@ def dashboard
     respond_with data, represent_with: Api::V1::Courses::DashboardRepresenter
   end
 
+  api :DELETE, '/courses/:course_id/teachers/:role_id', 'Removes a teacher from a course'
+  description <<-EOS
+  EOS
+  def remove_teacher
+    course = Entity::Course.find(params[:id])
```
Comment (jpslav) [inline app/controllers/api/v1/courses_controller.rb]: Can you make a teachers controller (similar to students controller) to handle deletion (and later addition, might also soon handle teacher self registration)?  Then this remove_teacher becomes the normal `destroy` action in that controller.

File: app/access_policies/course_access_policy.rb [hunk 2]
```diff
@@ -9,6 +9,10 @@ def self.action_allowed?(action, requestor, course)
        UserIsCourseTeacher[user: requestor, course: course])
     when :export, :roster, :add_period
```
Comment (jpslav) [inline app/access_policies/course_access_policy.rb]: Coincidentally, I think this `:roster` action is no longer used (moved to students controller).  Can you move?

File: app/controllers/api/v1/courses_controller.rb (hunk 1 above)
Comment (karenc) [inline app/controllers/api/v1/courses_controller.rb]: Yes.

File: app/access_policies/course_access_policy.rb (hunk 2 above)
Comment (karenc) [inline app/access_policies/course_access_policy.rb]: We still use this permission in students controller: https://github.com/openstax/tutor-server/blob/bb2b4c9240059ad50300f136d61dcb4c813e47fa/app/controllers/api/v1/students_controller.rb#L21

What would you like me to do?

**Comment by jpslav** [comment]
Perfect, thanks @karenc!

**Pull request merged**